sdks/java/harness/src/test/java/org/apache/beam/fn/harness/FnHarnessTest.java

```diff
@@ -56,13 +67,39 @@
           .setRegister(BeamFnApi.RegisterResponse.getDefaultInstance())
           .build();
 
+  private static @Mock Runnable onStartupMock = mock(Runnable.class);
+  private static @Mock Consumer<PipelineOptions> beforeProcessingMock = mock(Consumer.class);
+
+  /**
+   * Fake BeamWorkerInitializer that simply forwards calls to mocked functions so that they can be
+   * observed in tests.
+   */
+  @AutoService(BeamWorkerInitializer.class)
+  public static class FnHarnessTestInitializer extends BeamWorkerInitializer {
+    @Override
+    public void onStartup() {
+      onStartupMock.run();
+    }
+
+    @Override
+    public void beforeProcessing(PipelineOptions options) {
+      beforeProcessingMock.accept(options);
+    }
+  }
+
   @Test(timeout = 10 * 1000)
   @SuppressWarnings("FutureReturnValueIgnored") // failure will cause test to timeout.
   public void testLaunchFnHarnessAndTeardownCleanly() throws Exception {
+    Function<String, String> environmentVariableMock = mock(Function.class);
+
     PipelineOptions options = PipelineOptionsFactory.create();
 
+    when(environmentVariableMock.apply("HARNESS_ID")).thenReturn("id");
```

Comment:
Does mockito add anything here? A simple lambda is clear

Reply:
Mockito just gives me an easy way to verify the order in which things occurred. I decided to use a mock for the environment variable accessor so I could use reading it as a proxy for "starting to do anything other than running BeamWorkerInitialzier.onStartup implementations."  That way I can assert the proper ordering occured: `onStartup` called, worker initialization (signaled by accessing environment), `beforeProcessing` called, start processing data.

I'm definitely fooling myself a bit here, this doesn't really assert that `onStartup` is _the first_ thing that happened, just that it happened before any initialization based on environment variables. Do you think it would be better to just use a lambda for the environment variable accessor, and just assert the order `onStartup` -> `beforeProcessing` -> process data?